python/pyarrow/tests/test_parquet.py

```diff
@@ -284,6 +284,12 @@ def test_special_chars_filename(tempdir, use_legacy_dataset):
     assert table_read.equals(table)
 
 
+@pytest.mark.slow
+def test_file_with_over_int16_max_row_groups():
```

Comment:
Is this test supposed to succeed? Or is the 16-bit limitation only for encrypted files?

Reply:
Correct only for encrypted files. So this worked prior to the encrypted patch landing which introduced the artificial limitation for unencrypted files